src/mlpack/methods/reinforcement_learning/ppo_impl.hpp

```diff
@@ -155,20 +156,17 @@ void PPO<
     arma::mat dratio2 = (ratio >= (1 - config.Epsilon())) %
         (ratio <= (1 + config.Epsilon())) % advantages % dsurro;
 
-    arma::mat dprob = (dratio1 + dratio2) %
-        arma::exp(vectorise(prob - oldProb, 1));
-
-    arma::mat dmu = (vectorise(observation, 1) - mu) /
-        (arma::square(sigma)) % dprob;
+    arma::mat dprob = (dratio1 + dratio2) % ratio;
 
-    arma::mat dsigma = -1 / sigma +
-        arma::square(vectorise(observation, 1) - mu) / arma::pow(sigma, 3);
+    arma::mat dmu = (observation.t() - mu) / (arma::square(sigma)) % dprob;
+    arma::mat dsigma = -1.0 / sigma +
+        arma::square(observation.t() - mu) / arma::pow(sigma, 3);
 
     arma::mat dTanh, dSoftP;
-    ann::TanhFunction::Deriv(vectorise(dmu, 1), dTanh);
-    ann::SoftplusFunction::Deriv(vectorise(dsigma, 1), dSoftP);
+    ann::TanhFunction::Deriv(dmu.t(), dTanh);
```

Comment:
If you don't transpose ``dmu, dsigma`` you don't have to transpose it in the ``join_cols(..)`` method.

Reply:
> If you don't transpose `dmu, dsigma` you don't have to transpose it in the `join_cols(..)` method.

Oh, yes, such a silly mistake. Fixed.